File: 백준 Python/Step7 함수/pro7_3.py
import sys
n = int(sys.stdin.readline());
hanCount = 0;
for test in range(1, n+1) : #한수인지 테스트해볼 숫자 test
    each =[];
    big = len(str(test)); #몇자리수인지 구하기.
    for i in range(big-1,-1,-1) : #자리수별로 찢기. 4자리수라면 3부터 -1까지로 범위를 정해줘야 인덱스 3 2 1 0 로 자리에 맞게 4번 실행됨.
        each.append(int((test/10**i)%10));
       # print(each)
    han = False;
    if big == 1 or big == 2 : #한자리수,두자리수면 모두 한수.
        han = True;
        #print('한자리수, 두자리수=한수')
    else :
        for i in range(big-2) : #몇번 비교할지. 세자리수면 1번, 네자리수면 2번만 하면 비교하면 됨.
            if each[i]-each[i+1] == each[i+1]-each[i+2] :
                han = True;
                #print('한수통과',test,i,' : ',han)
            else :
                han = False;
                #print('한수아님.break',test,i,' : ',han)
                break; # 하나라도 같지 않으면 등차수열이 아니므로 반복문 탈출.
    if han == True :
        hanCount += 1;
        #print(hanCount)
print(hanCount);

File: 백준 Java/Step7_old 함수/src/Pro7_3.java
import java.io.*;

public class Pro7_3 {
	public static void main(String[] args) throws IOException{
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int hanNum = 0;
		int n = Integer.parseInt(br.readLine());
		for(int test=1; test<n+1; test++) {
			int big = (int)(Math.log10(test)+1); //test가 몇자리 수인지 구함
			//System.out.println(test+"의 자리수는"+big);
			int[] each = new int[big];
			for(int i=0; i<big; i++) {
				each[i] = (int)((test/Math.pow(10, big-i-1))%10);
				//System.out.println("각 자리의 숫자는 "+(int)((test/Math.pow(10, big-i-1))%10));
			}
			boolean han = false;
			if(big == 1 || big ==2) {
				han = true;
			}
			else {
				for(int i=0; i<big-2; i++) {
					if(each[i]-each[i+1] == each[i+1]-each[i+2]) {
						han = true;
						for(int k=0; k<big; k++) {
							//System.out.println(test+"일 때"+k+"번째 자릿수는 "+each[k]);
						}
						//System.out.println((each[i]-each[i+1])+" == "+(each[i+1]-each[i+2]));
						//System.out.println(test+"일 때 한수true");
					}
					else {
						//System.out.println(test+"일 때 한수false");
						break;
						
					}
				}
			}
			if(han == true) {
				hanNum += 1;
			}
		}
		System.out.println(hanNum);

	}

}
